src/background/set-timeout-manager.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { AlarmUtils } from 'background/alarm-utils';
+import { WindowUtils } from 'common/window-utils';
+
+export class SetTimeoutManager {
+    constructor(private utils: { alarmUtils?: AlarmUtils; windowUtils?: WindowUtils }) {}
```

Comment:
It's a little unclear to me when/why AlarmUtils or WindowUtils would be undefined. Is WindowUtils a temporary measure during the mv3 transition, or will we need to use this class in a context where window is defined? If we need both paths long-term, could this class be separated into something like BackgroundSetTimeoutManager and BrowserSetTimeoutManager?

Reply:
Oh that's a good point. I'll split this in two similar to how the `PromiseFactory` stuff is structured.